File: docs/deliverables/video/_work/hover-s2.py
import subprocess, json, sys

code_js = """async (page) => {
  // Find Claude S2 message
  const claudeMsg = page.locator('p:has-text("Hero")').first();
  await claudeMsg.scrollIntoViewIfNeeded();
  await page.waitForTimeout(300);

  // Find message container
  const msg = claudeMsg.locator('xpath=ancestor::div[2]');
  const cnt = await msg.count();
  if (cnt === 0) return { error: 'no message found' };

  // BEFORE hover
  const before = await page.evaluate(() => Array.from(document.querySelectorAll('button')).length);
  const beforeBtns = await page.evaluate(() => Array.from(document.querySelectorAll('button')).map(b => (b.title || b.getAttribute('aria-label') || b.innerText || '').substring(0, 40)).filter(t => t));

  // Use playwright .hover() API
  await msg.hover();
  await page.waitForTimeout(1000);

  // AFTER hover
  const after = await page.evaluate(() => Array.from(document.querySelectorAll('button')).length);
  const afterBtns = await page.evaluate(() => Array.from(document.querySelectorAll('button')).map(b => (b.title || b.getAttribute('aria-label') || b.innerText || '').substring(0, 40)).filter(t => t));

  const newBtns = afterBtns.filter(b => !beforeBtns.includes(b));
  return { before, after, newBtns, msgBox: await msg.boundingBox() };
}"""

args = {"code": code_js}
result = subprocess.run(
    [r"C:\Users\yhn\.mavis\bin\mavis.cmd", "mcp", "call", "playwright", "browser_run_code", "--stdin"],
    input=json.dumps(args, ensure_ascii=False),
    capture_output=True,
    text=True,
    encoding="utf-8",
)
print("STDOUT:", result.stdout[:2500])
print("STDERR:", result.stderr[:300])

File: docs/deliverables/video/_work/hover-s2.js
async (page) => {
  // Hover 第一条群聊消息看 Pin/复制代码按钮是否出现
  const msg = page.locator('article').first();
  if ((await msg.count()) === 0) return { error: 'no article' };

  // Before hover
  const before = await page.evaluate(() => {
    const all = Array.from(document.querySelectorAll('button'));
    return all.map(b => (b.title || b.getAttribute('aria-label') || b.innerText || '').substring(0, 30)).filter(t => t).slice(0, 30);
  });

  // Hover message center
  const box = await msg.boundingBox();
  if (!box) return { error: 'no bbox' };
  await page.mouse.move(box.x + box.width / 2, box.y + box.height / 2);
  await page.waitForTimeout(500);

  // After hover
  const after = await page.evaluate(() => {
    const all = Array.from(document.querySelectorAll('button'));
    return all.map(b => (b.title || b.getAttribute('aria-label') || b.innerText || '').substring(0, 30)).filter(t => t).slice(0, 30);
  });

  // New buttons added
  const newBtns = after.filter(b => !before.includes(b));
  return { before: before.length, after: after.length, newBtns, msgBox: { x: box.x, y: box.y, w: box.width, h: box.height } };
}
